api/src/client/ApiTaxFilingClient.test.ts

```diff
@@ -0,0 +1,237 @@
+/* eslint-disable unicorn/no-null */
+import { randomInt } from "@shared/intHelpers";
+import axios from "axios";
+
+import { generateTaxFilingResults, generateTaxIdAndBusinessNameAndEmail } from "../../test/factories";
+import { ApiTaxFilingClient } from "../domain/types";
+import { LogWriter, LogWriterType } from "../libs/logWriter";
+
+import { dateToShortISO } from "../domain/tax-filings/taxIdHelper";
+import {
+  apiTaxFilingClient,
+  ApiTaxFilingLookupRequest,
+  ApiTaxFilingLookupResponse,
+  ApiTaxFilingOnboardingRequest,
+  ApiTaxFilingOnboardingResponse,
+} from "./ApiTaxFilingClient";
+
+jest.mock("axios");
+jest.mock("winston");
+const mockAxios = axios as jest.Mocked<typeof axios>;
+
+const generateApiTaxFilingLookupResponse = (
+  overrides: Partial<ApiTaxFilingLookupResponse>,
+  state: "PENDING" | "SUCCESS" | "FAILED" = "SUCCESS"
+): ApiTaxFilingLookupResponse => ({
+  ApiKey: `some-ApiKey-${randomInt()}`,
+  Data:
+    state == "SUCCESS"
+      ? [
+          { Name: "taxpayer-id", Value: randomInt(9).toString() },
+          { Name: "taxpayer-location", Value: randomInt(3).toString() },
+          { Name: "tax-business-name", Value: `Some, Name-${randomInt()}` },
+          { Name: "tax-name-control", Value: "Some" },
+        ]
+      : null,
+  Errors:
+    state != "SUCCESS"
+      ? [
+          {
+            Error:
+              state == "PENDING"
+                ? "There is no specific tax eligibility for this business at the moment but please do come back to check later since we get weekly updates."
+                : "No matching record found for the business. Please verify your Taxpayer ID and Business Name.",
+            Field: "Results",
+          },
+        ]
+      : null,
+  Results: state == "SUCCESS" ? generateTaxFilingResults() : null,
+  ...overrides,
+});
+
+const generateApiTaxFilingOnboardingResponse = (
+  overrides: Partial<ApiTaxFilingOnboardingResponse>,
+  statusCode: 200 | 400 | 500 | number = 200
+): ApiTaxFilingOnboardingResponse => {
+  return {
+    ApiKey: `some-ApiKey-${randomInt()}`,
+    Errors:
+      statusCode == 400
+        ? [
+            {
+              Error: `some-error-content-${randomInt()}`,
+              Field: randomInt() % 2 === 0 ? "Taxpayer ID" : "Business Name",
+            },
+          ]
+        : [],
+    Notice: statusCode == 200 ? "You are signed up for the Tax Calendar Reminder Service." : null,
+    StatusCode: statusCode as 200 | 400 | 500,
+    ...overrides,
+  };
```

Comment:
there's a lot of logic in these two generators. It might be worth considering simplifying this to something like `generate400OnboardingResponse` and `generate200OnboardingResponse` to make it clearer what happens when. Might make it easier for someone to edit in the future. 🤷